openvdb_cmd/vdb_tool/include/Parser.h

```diff
@@ -0,0 +1,1041 @@
+
+#include <iostream>
+#include <sstream>
+#include <string> // for std::string, std::stof and std::stoi
+#include <algorithm> // std::sort
+#include <random>
+#include <functional>
+#include <vector>
+#include <list>
+#include <set>
+#include <time.h>
+#include <initializer_list>
+#include <unordered_map>
+#include <iterator>// for std::advance
+#include <sys/stat.h>
+#include <stdio.h>
+
+#include <openvdb/openvdb.h>
+
+#include "Util.h"
+
+namespace openvdb {
+OPENVDB_USE_VERSION_NAMESPACE
+namespace OPENVDB_VERSION_NAME {
+namespace vdb_tool {
+
+// ==============================================================================================================
+
+/// @brief This class defines string attributes for options, i.e. arguments for actions
+struct Option {
+    std::string name, value, example, documentation;
+    void append(const std::string &v) {value = value.empty() ? v : value + "," + v;}
+};
+
+// ==============================================================================================================
+struct Action {
+    std::string            name;// primary name of action, eg "read"
+    std::string            alias;// alternate name for action, eg "i"
+    std::string            documentation;// documentation e.g. "read", "i", "files", "read files"
+    size_t                 anonymous;// index of the option to which the value of un-named option will be appended, e.g. files
+    std::vector<Option>    options;// e.g. {{"grids", "density,sphere"}, {"files", "path/base.ext"}}
+    std::function<void()>  init, run;// callback functions
+
+    /// @brief c-tor
+    Action(std::string _name,
+           std::string _alias,
+           std::string _doc,
+           std::vector<Option> &&_options,
+           std::function<void()> &&_init,
+           std::function<void()> &&_run,
+           size_t _anonymous = -1)
+      : name(std::move(_name))
+      , alias(std::move(_alias))
+      , documentation(std::move(_doc))
+      , anonymous(_anonymous)
+      , options(std::move(_options))
+      , init(std::move(_init))
+      , run(std::move(_run)) {}
+    /// @brief default copy constructor
+    Action(const Action&) = default;
+    /// @brief Sets the options of this actions
+    void setOption(const std::string &str);
+    void print(std::ostream& os = std::cerr) const;
+};// Action struct
+
+// ==============================================================================================================
+
+using ActListT = std::list<Action>;
+using ActIterT = typename ActListT::iterator;
+using VecF = std::vector<float>;// vector of floats
+using VecI = std::vector<int>;// vector for integers
+using VecS = std::vector<std::string>;// vector of strings
+
+// ==============================================================================================================
+
+/// @brief Class that stores values by name
+class Memory
+{
+    std::unordered_map<std::string, std::string> mData;
```

Comment:
As a general comment, the member variables are usually included at the end of the class definition.